x-pack/legacy/plugins/apm/public/components/shared/Links/DiscoverLinks/DiscoverLink.tsx

```diff
@@ -32,18 +31,13 @@ interface Props {
 
 export function DiscoverLink({ query = {}, ...rest }: Props) {
   const core = useKibanaCore();
-  const { apmIndexPattern } = useAPMIndexPattern();
   const location = useLocation();
 
-  if (!apmIndexPattern.id) {
-    return null;
-  }
-
   const risonQuery = {
     _g: getTimepickerRisonData(location.search),
     _a: {
       ...query._a,
-      index: apmIndexPattern.id
+      index: 'apm-*' // index pattern id specified in `/src/legacy/core_plugins/kibana/server/tutorials/apm/index_pattern.json`
```

Comment:
Hardcode `apm-*` as the id. This is not a matcher, just an identifier. We could call it `apm-app-index-pattern-id` if we wanted to (that might actually not be a bad idea)